srt-protocol/src/protocol/mod.rs

```diff
@@ -12,30 +13,42 @@ pub mod stats;
 
 /// Timestamp in us after creation
 /// These wrap every 2^32 microseconds
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Copy, Clone, PartialEq, Eq)]
 pub struct TimeStamp(Wrapping<u32>);
 
 /// Signed duration in us, e.g. RTT
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Ord, PartialOrd)]
+#[derive(Copy, Clone, PartialEq, Eq, Ord, PartialOrd)]
 pub struct TimeSpan(i32);
 
-const TIMESTAMP_MASK: u128 = u32::MAX as u128;
-const MAX_DIFF_US: u32 = u32::MAX / 2;
-const WRAP_US: u64 = (u32::MAX as u64) + 1;
-
 #[derive(Copy, Clone, Debug)]
-pub struct TimeBase(Instant);
+pub struct TimeBase {
+    origin_time: Instant,
+    reference_time: Instant,
+    reference_ts: TimeStamp,
```

Comment:
Would be great if you had some comments here on what the various fields meant

It seems like: origin_time is the old .0
reference_time: is the time of the last wrap? Or something?

Reply:
Yes, origin_time is .0 or "start_time" in other contexts. I'm open to alternative name suggestions.
The "reference" fields are equivalent time points from their respective time scales that are reference points for mapping between an Instance and a sender TimeStamp.

I'll add this comment to the code.